File: server/controller/logic/article.go
package logic

import (
	"context"
	"encoding/json"
	"webback/db"

	"github.com/redis/go-redis/v9"
)

func GetArticle(db_article *db.Article) Article {
	article := Article{
		Model:     db_article.Model,
		Title:     db_article.Title,
		Content:   db_article.Content,
		Type:      db_article.Type,
		PageViews: db_article.PageViews,
	}

	// 获取分类
	article.Category.ID = db_article.CategoryID
	article.Category.Name = db.GetCategory(db_article.CategoryID).Name

	// 获取作者
	article.User.ID = db_article.UserID
	article.User.Name = db.GetUser(db_article.UserID).Name

	// 获取标签
	tags := db.GetTagsByArticleID(db_article.ID)
	article.Tags = make([]Tag, len(tags))
	for i, tag := range tags {
		article.Tags[i].ID = tag.ID
		article.Tags[i].Name = tag.Name
	}

	return article
}

func (articles Articles) LoadIntoRedis() (err error) {
	_, err = db.RDB.Del(context.Background(), "article:all").Result()
	if err != nil {
		return err
	}
	for _, article := range articles {
		json, err := json.Marshal(article)
		if err != nil {
			return err
		}
		_, err = db.RDB.ZAdd(context.Background(), "article:all", redis.Z{
			Score:  float64(article.CreatedAt.Unix()),
			Member: json,
		}).Result()
		if err != nil {
			return err
		}
	}
	return
}

func (articles *Articles) GetFromRedis() (err error) {
	res, err := db.RDB.ZRevRange(context.Background(), "article:all", 0, -1).Result()
	if err != nil {
		return err
	}

	*articles = make(Articles, len(res))
	for i, v := range res {
		err = json.Unmarshal([]byte(v), &(*articles)[i])
		if err != nil {
			return err
		}
	}
	return nil
}

File: front/src/store/article.ts
import api from "@/api/api";
import { reactive, ref } from "vue";

export interface Article {
  ID: number;
  CreatedAt: string;
  ModifiedAt: string;
  Category: Category;
  User: User;
  Title: string;
  Content: string;
  Type: number; // 0-原创 1-转载
  Tags: Tag[];
  PageViews: number;
  Likes: number; // 点赞数
}
export interface Category {
  ID: number;
  Name: string;
}
export interface User {
  ID: number;
  Name: string;
}
export interface Tag {
  ID: number;
  Name: string;
}
export interface Comment {
  ID: number;
  CreatedAt: string;
  User: User;
  ArticleID: number;
  ReplyTopCommentID: number;
  ReplyCommentID: number;
  Content: string;
  Favor: number;

  Floor: number;
}

const defaultData: Article = Object.freeze({
  ID: 0,
  CreatedAt: "",
  ModifiedAt: "",
  Category: {
    ID: 0,
    Name: "",
  },
  User: {
    ID: 0,
    Name: "",
  },
  Title: "",
  Content: "",
  Type: 0,
  Tags: [],
  PageViews: 0,
  Likes: 0,
});

const list: Article[] = reactive([]);

const allCategory: Category[] = reactive([]);
const getAllCategory = async () => {
  try {
    const res = await api.getCategory("all");
    Object.assign(allCategory, res.data);
  } catch (err) {
    console.error(err);
  }
};
const allTag: Tag[] = reactive([]);
const getAllTag = async () => {
  try {
    const res = await api.getTag("all");
    Object.assign(allTag, res.data);
  } catch (err) {
    console.error(err);
  }
};

export { defaultData, list, allCategory, getAllCategory, allTag, getAllTag };
